python/tests/transform_coords_test.py

```diff
@@ -51,10 +51,6 @@ def ab(*, a, b):
     return a + b
 
 
-def ba(b, a):
```

Comment:
Why was this one removed?

Reply:
This was part of the workaround #2086 which I reverted as part of this PR, since it it not necessary any more.

Reply:
But the test still has to pass in order to implement robust unit conversions.

Reply:
True, but isn't it a case of not re-testing basic assumptions (which was fixed here)? Also, "transposed" coords are now exercised by the test below where a TODO was removed.

Reply:
Not sure if this is re-testing something. I would say we should have a test that ensures that the order of dimensions of newly computed coordinates does not matter in `transform_coords`.

Reply:
But wouldn't it be better to have just tests that ensure that coords with arbitrary dim order can be added to a data array, rather than re-testing this contract (of `DataArray`) every time we implement something that relies on this contract?

Reply:
> But wouldn't it be better to have just tests that ensure that coords with arbitrary dim order can be added to a data array

Isn't this what the test with `ba` did? Yes, the order is arbitrary in all tests, but it is always the same.

Reply:
@jl-wynen I made the corresponding unit test more thorough now.